File: proxy/tests/script_engine/query_clear.js
/// <reference path="../../script_libs/js/roxy.d.ts" />
/** @type {Extension} */
const queryClear = {
  request(flow) {
    if (flow.request.url.searchParams.get("foo") == "bar") {
      flow.request.url.searchParams.clear();
    }
  }
};
globalThis.extensions = [queryClear];

File: proxy/tests/script_engine/query_clear.lua
pcall(require, "../../script_libs/lua/roxy.lua")
---@type Extension
local query_clear = {
	request = function(flow)
		if flow.request.url.search_params["foo"] == "bar" then
			flow.request.url.search_params:clear()
		end
	end,
}
Extensions = { query_clear }
